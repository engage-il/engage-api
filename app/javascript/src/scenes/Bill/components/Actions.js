// @flow
import * as React from 'react'
import { createFragmentContainer, graphql } from 'react-relay'
import { Button } from '@/components'
import type { Bill } from '@/types'

type Props = {
  bill: Bill
}

let Actions = function Actions ({ bill }: Props) {
  return (
    <div>
      <div>
        <Button
          to={bill.witnessSlipUrl}
          children='Take Action'
        />
        <Button
          to={bill.witnessSlipResultUrl}
          isSecondary
          children='View Results'
        />
      </div>
      <div>
        <Button
          to={bill.detailsUrl}
          isSecondary
          children='View Details'
        />
        <Button
          to={bill.fullTextUrl}
          isSecondary
          children='View Full Text'
        />
      </div>
    </div>
  )
}

Actions = createFragmentContainer(Actions, graphql`
  fragment Actions_bill on Bill {
    detailsUrl
    fullTextUrl
    witnessSlipUrl
    witnessSlipResultUrl
  }
`)

export { Actions }
