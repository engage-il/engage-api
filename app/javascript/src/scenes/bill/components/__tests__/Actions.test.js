/* eslint-env jest */
import * as React from 'react'
import { shallow } from 'enzyme'
import { defaultsDeep } from 'lodash'
import { Actions } from '../Actions'

// subject
let subject

const defaults = {
  bill: {
    witnessSlipUrl: 'http://www.test.com/slip',
    witnessSlipResultUrl: 'http://www.test.com/result',
    billDetailsUrl: 'http://www.test.com/details',
    fullTextUrl: 'http://www.test.com/text'
  }
}

function loadSubject (props = {}) {
  subject = shallow(<Actions {...defaultsDeep(props, defaults)} />)
}

// specs
afterEach(() => {
  subject = null
})

describe('#render', () => {
  it('shows the actions', () => {
    loadSubject()
    expect(subject).toMatchSnapshot()
  })
})
