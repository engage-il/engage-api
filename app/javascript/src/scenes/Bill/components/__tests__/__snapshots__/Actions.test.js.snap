// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render shows the actions 1`] = `
<div
  data-css-2jmw69=""
>
  <div>
    <Button
      styles={
        Object {
          "data-css-1h39hs6": "",
        }
      }
      to="http://www.test.com/slip"
    >
      Take Action
    </Button>
    <Button
      isSecondary={true}
      styles={
        Object {
          "data-css-1h39hs6": "",
        }
      }
      to="http://www.test.com/result"
    >
      View Results
    </Button>
  </div>
  <div>
    <Button
      isSecondary={true}
      styles={
        Object {
          "data-css-1h39hs6": "",
        }
      }
    >
      View Details
    </Button>
    <Button
      isSecondary={true}
      styles={
        Object {
          "data-css-1h39hs6": "",
        }
      }
      to="http://www.test.com/text"
    >
      View Full Text
    </Button>
  </div>
</div>
`;
